
  <launch>

  <arg name="ticks_per_meter" value="13995" />
  <arg name="wheel_separation" value="0.2" />
  <arg name="base_frame_id" value="base_footprint" />
  <arg name="encoder_min" value="-65535" />
  <arg name="encoder_max" value="65535" />

  <arg name="port" value="/dev/ttyACM0" />
  <arg name="baudrate" value="115200" />
  <arg name="imu_frame" value="imu_link" />

  <node name="diff_drive_controller" pkg="diff_drive" type="diff_drive_controller"
        output="screen">
    <rosparam subst_value="true">
      ticks_per_meter: $(arg ticks_per_meter)
      wheel_separation: $(arg wheel_separation)
      max_motor_speed: 3250
      timeout: 1.0
    </rosparam>
    <remap from="cmd_vel" to="/robot/cmd_vel" />
  </node>
  
  <node name="odom_publisher" pkg="diff_drive" type="diff_drive_odometry"
        output="screen">
    <rosparam subst_value="true">
      ticks_per_meter: $(arg ticks_per_meter)
      wheel_separation: $(arg wheel_separation)
      base_frame_id: $(arg base_frame_id)
      encoder_min: $(arg encoder_min)
      encoder_max: $(arg encoder_max)
    </rosparam>
  </node>

  <node pkg="realsense_explorer_control" type="serial_connection.py" name="serial_connection_RobotHW" output="screen">
    <rosparam subst_value="true">
      port: $(arg port)
      baudrate: $(arg baudrate)
      imu_frame: $(arg imu_frame)
    </rosparam>
  </node>

    <node pkg="realsense_explorer_control" type="servokit_control_zmq.py" 
          name="jointState_to_servo" output="screen"/>

  <node pkg="realsense_explorer_control"
          type="run_servo_zmq.sh" name="zmq_servo_com"
          output="screen"
  />


  <!-- <node name="controller" pkg="pid" type="controller" ns="left_wheel" output="screen" >
    <param name="Kp" value="5.0" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.1" />
    <param name="upper_limit" value="255" />
    <param name="lower_limit" value="-255" />
    <param name="windup_limit" value="255" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="100.0" />
    <remap from="setpoint" to="/l_wheel_setpoint" />
    </node>


  <node name="controller" pkg="pid" type="controller" ns="right_wheel" output="screen" >
    <param name="Kp" value="-4.0" />
    <param name="Ki" value="-0.0" />
    <param name="Kd" value="-0.3" />
    <param name="upper_limit" value="255" />
    <param name="lower_limit" value="-255" />
    <param name="windup_limit" value="255" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="100.0" />
    <remap from="setpoint" to="/r_wheel_setpoint" />
  </node> -->


    
  <!-- <node pkg="robot_localization" type="ekf_localization_node" name="robot_ekf_localization" clear_params="true">
    <rosparam command="load" file="$(find realsense_explorer_control)/config/ekf_config.yaml" />
  </node> -->
    
  </launch>